EvalServer/src/controllers/deepeval.py

```
@@ -812,22 +812,29 @@ async def read_deepeval_dataset_controller(path: str) -> JSONResponse:
     - Relative to datasets folder (e.g., "chatbot/chatbot_basic.json")
     """
     try:
+        # Security: Early rejection of suspicious path patterns
+        if ".." in path or path.startswith("/") or path.startswith("\\"):
+            raise HTTPException(status_code=400, detail="Invalid dataset path: path traversal not allowed")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)